src/txmempool.cpp

```diff
@@ -404,6 +404,13 @@ bool CTxMemPool::addUnchecked(const uint256& hash, const CTxMemPoolEntry &entry,
         }
     }
 
+    // Save spent nullifiers
+    if (tx.IsShieldedTx()) {
+        for (const SpendDescription& sd : tx.sapData->vShieldedSpend) {
+            mapSaplingNullifiers[sd.nullifier] = &tx;
```

Comment:
nit: could use `TransactionRef` for this. Changing `mapSaplingNullifiers` and here doing a newit->GetSharedTx()`